src/sonic_ax_impl/mibs/vendor/cisco/ciscoEntityFruControlMIB.py

```diff
@@ -18,35 +51,59 @@ def __init__(self):
         """
         init the handler
         """
-        self.psuutil = None
+        self.statedb = SonicV2Connector()
+        self.statedb.connect(self.statedb.STATE_DB)
 
-        try:
-            module = imp.load_source(PSU_PLUGIN_MODULE_NAME, PSU_PLUGIN_MODULE_PATH)
-        except ImportError as e:
-            mibs.logger.error("Failed to load PSU module '%s': %s" % (PSU_PLUGIN_MODULE_NAME, str(e)), True)
-            return
-        except FileNotFoundError as e:
-            mibs.logger.error("Failed to get platform specific PSU module '%s': %s" % (PSU_PLUGIN_MODULE_NAME, str(e)), True)
-            return
+        self.psu_name_prefix = 'PSU '
+        self.chassis_name = 'chassis 1'
 
-        try:
-            platform_psuutil_class = getattr(module, PSU_PLUGIN_CLASS_NAME)
-            self.psuutil = platform_psuutil_class()
-        except AttributeError as e:
-            mibs.logger.error("Failed to instantiate '%s' class: %s" % (PLATFORM_SPECIFIC_CLASS_NAME, str(e)), True)
+        self.psu_presence_ok = 'true'
+        self.psu_status_ok = 'true'
+
+    def _getPsuNum(self):
+        """
+        Get PSU number
+        :return: the number of supported PSU
+        """
+        chassis_info = self.statedb.get_all(self.statedb.STATE_DB, mibs.chassis_info_table(self.chassis_name))
+        psu_num = get_chassis_data(chassis_info)
+
+        return int(psu_num[0])
+
+    def _getPsuPresence(self, psu_index):
+        """
+        Get PSU presence
+        :return: the presence of particular PSU
+        """
+        psu_name = self.psu_name_prefix + str(psu_index)
+        psu_info = self.statedb.get_all(self.statedb.STATE_DB, mibs.psu_info_table(psu_name))
+        presence, status = get_psu_data(psu_info)
+
+        return presence == self.psu_presence_ok
+
+    def _getPsuStatus(self, psu_index):
+        """
+        Get PSU status
+        :return: the status of particular PSU
+        """
+        psu_name = self.psu_name_prefix + str(psu_index)
+        psu_info = self.statedb.get_all(self.statedb.STATE_DB, mibs.psu_info_table(psu_name))
+        presence, status = get_psu_data(psu_info)
+
+        return status == self.psu_status_ok
 
     def _getPsuIndex(self, sub_id):
         """
         Get the PSU index from sub_id
         :return: the index of supported PSU
         """
-        if not self.psuutil or not sub_id or len(sub_id) > 1:
+        if not sub_id or len(sub_id) > 1:
             return None
 
         psu_index = int(sub_id[0])
 
         try:
-            num_psus = self.psuutil.get_num_psus()
+            num_psus = self._getPsuNum()
         except Exception:
             # Any unexpected exception or error, log it and keep running
             mibs.logger.exception("PowerStatusHandler._getPsuIndex() caught an unexpected exception during get_num_psus()")
```

Comment:
we are using "_getPsuNum()" now, "get_num_psus()" should be replaced. Have similiar case in below code.